lib/Result.ts

```diff
@@ -161,6 +183,28 @@ export class FailureResult<Success, Failure> {
     this.observe(handler)
     return this
   }
+  
+  /**
+   * Runs the handler with the current success value and passes the current success value through
+   * if the handler is successful.
+   *
+   * @param handler a function to process the current success value and either passthrough the value or return a failure.
+   */
+  passthroughSuccess<NewSuccess, NewFailure>(_: (value: Success) => AwaitableResult<NewSuccess, NewFailure>) {
+    return this as unknown as FailureResult<Success, NewFailure | Failure>
+  }
+  
+  /**
+   * Runs the handler with the current failure value and passes the current failure value through
+   * if the handler is successful.
+   *
+   * @param handler a function to process the current failure value and either passthrough the value or return a failure.
+   */
+  passthroughFailure<NewSuccess, NewFailure>(
+    handler: (value: Failure) => AwaitableResult<NewSuccess, NewFailure>
+  ): FailureResult<Success, NewFailure | Failure> {
```

Comment:
```suggestion
  ): AwaitableResult<Success, NewFailure | Failure> {
```
See same comment on `SuccessResult`.

Reply:
fixed